filebeat/config/config.go

```diff
@@ -86,6 +87,12 @@ type MultilineConfig struct {
 	Timeout  string `yaml:"timeout"`
 }
 
+type JsonDecoderConfig struct {
+	OnUnmarshalError string `yaml:"on_unmarshal_error"`
```

Comment:
I would like to see this field validated at startup to guarantee that it is `ignore`, `add_error_key`, or empty. This way the user gets an error instead of it just defaulting to `ignore`.